identity_vc/src/common/oom.rs

```diff
@@ -0,0 +1,109 @@
+use std::fmt;
+
+/// A generic container that stores one or many values of a given type.
+#[derive(Clone, Hash, PartialEq, Eq, PartialOrd, Ord, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum OneOrMany<T> {
```

Comment:
You could write a custom Serializer/Deserializer to handle the cases where you get only one piece of data as opposed to a larger value.  That's what I did for the Context in the DIDDocument.  If the vector only contains one value, represent it as a single string in the Json and if it contains more than one value, represent it as a list.  In this case, it would be one String vs Multiple nested objects.